### PR TITLE
Query log rotation

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Add possibility to rotate query. Rotation threshold is specified by 'dbms.querylog.rotation.threshold' property.
By default rotation will happen each 20m.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
<b>Note</b> should be null merged. 2.3 have separate PR

**Comment by cleishm** [comment]
Rather than configuring the query logging within the kernel, I wonder if we should instead expose a method by which an embedded user can inject their own `QueryLogger` -- similarly to how they can provide a `LogProvider` for getting user message logging?

Obviously the server would provide the logger that writes to a specific file (which is fine since it has a known directory layout to follow), along with all the configuration for rotation, etc.

**Comment by MishaDemianenko** [comment]
@cleishm have added separate card into backlog for that.

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 1]
```diff
@@ -406,6 +406,10 @@ private static String defaultPageCacheMemory()
             "provided query logging is enabled. Defaults to 0 seconds, that is all queries are logged.")
     public static final Setting<Long> log_queries_threshold = setting("dbms.querylog.threshold", DURATION, "0s");
 
+    @Description( "Specifies at which file size the query log will auto-rotate. ")
+    public static final Setting<Long> log_queries_rotation_threshold = setting("dbms.querylog.rotation.threshold",
+            BYTES, "20m", min( 1024 * 1024L ), max( Long.MAX_VALUE ) );
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:411]: Where does this default come from? Is it reasonable under decent load?

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:411]: That is aligned with `logical_log_rotation_threshold`

**Pull request merged**